### PR TITLE
:seedling: migrate to kcp prow

### PR DESCRIPTION
**Pull request by palnabarun**
## Summary

Migrate to kcp-prow

## Related issue(s)

Fixes https://github.com/kcp-dev/infra/issues/27

/assign @xrstf

### CODE REVIEW

File: .prow.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    always_run: true
+    decorate: true
+    clone_uri: "ssh://git@github.com/kcp-dev/apimachinery.git"
+    labels:
+      preset-goproxy: "true"
+    spec:
+      containers:
+        # TODO(nabarun): Change the following to the kcp-dev/build image once it is available
+        # Ref: https://github.com/kcp-dev/infra/pull/34
+        - image: quay.io/kubermatic/build:go-1.20-node-18-6
```
Comment (xrstf) [inline .prow.yaml:27]: Please update to `ghcr.io/kcp-dev/infra/build:1.19.9-2` now :)

Comment (palnabarun) [reply .prow.yaml]: The PR is updated with the latest commits now. 🎉

**Comment by palnabarun** [comment]
@xrstf all tests are passing now. Can you please have a look?

**Review by xrstf — APPROVED** [review]
/lgtm

@sttts  PTAL

**Comment by xrstf** [comment]
/lgtm cancel

@palnabarun Can you also remove the CI Github workflows?

**Pull request merged**